src/libcrun/cgroup.c

```diff
@@ -736,7 +738,71 @@ systemd_job_removed (sd_bus_message *m, void *userdata, sd_bus_error *error arg_
     return -1;
 
   if (strcmp (p->path, path) == 0)
-    *p->terminated = 1;
+    {
+      p->terminated = 1;
```

Comment:
theoretically, we can get rid of `terminated`, but I am not sure if `unit` and `result` are guaranteed to be set.